src/Nethermind/Nethermind.DataMarketplace.Consumers/Shared/Services/ConsumerServicesBackgroundProcessor.cs

```diff
@@ -186,6 +182,18 @@ private async Task ProcessBlockAsync(long number, long timestamp)
             }
         }
 
+        private async Task<IEnumerable<DepositDetails>> GetDepositsToRefundAsync()
+        {
+            PagedResult<DepositDetails> depositsToRefundResult = await _depositManager.BrowseAsync(new GetDeposits
+            {
+                EligibleToRefund = true,
+                CurrentBlockTimestamp = _currentBlockTimestamp,
+                Results = int.MaxValue
+            });
+
+            return depositsToRefundResult.Items.Where(d => d.ConsumedUnits < d.Deposit.Units);
```

Comment:
shall this not be handled by the query properly? why are they marked as eligible for refund?